src/test/java/io/cryostat/util/URIRangeTest.java

```diff
@@ -0,0 +1,168 @@
+        "172.17.10.5, false",
+        "172.100.10.5, false",
+        "192.168.1.1, false",
+        "192.168.2.1, false",
+        "192.169.2.1, false",
+        "svc.local, false",
+        "example.svc.cluster.local, false",
+        "example.com, false"
+    })
+    public void testLinkLocalRange(String s, String v) throws URISyntaxException {
+        test(URIRange.LINK_LOCAL, s, v);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "127.0.0.1, true",
+        "localhost, true",
+        "svc.localhost, true",
+        "169.254.0.0, true",
+        "169.254.10.5, true",
+        "168.254.1.1, false",
+        "10.0.0.10, true",
+        "10.0.10.10, true",
+        "10.10.10.10, true",
+        "11.10.10.10, false",
+        "172.16.0.0, true",
+        "172.16.10.5, true",
+        "172.17.10.5, true",
+        "172.100.10.5, false",
+        "192.168.1.1, true",
+        "192.168.2.1, true",
+        "192.170.2.1, false",
+        "svc.local, false",
+        "example.svc.cluster.local, false",
+        "example.com, false"
+    })
+    public void testSiteLocalRange(String s, String v) throws URISyntaxException {
+        test(URIRange.SITE_LOCAL, s, v);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "127.0.0.1, true",
+        "localhost, true",
+        "svc.localhost, true",
+        "169.254.0.0, true",
+        "169.254.10.5, true",
+        "168.254.1.1, false",
+        "10.0.0.10, true",
+        "10.0.10.10, true",
+        "10.10.10.10, true",
+        "11.10.10.10, false",
+        "172.16.0.0, true",
+        "172.16.10.5, true",
+        "172.17.10.5, true",
+        "172.100.10.5, false",
+        "192.168.1.1, true",
+        "192.168.2.1, true",
+        "192.169.2.1, false",
+        "svc.local, true",
+        "example.svc.cluster.local, true",
+        "example.com, false"
+    })
+    public void testDnsLocalRange(String s, String v) throws URISyntaxException {
+        test(URIRange.DNS_LOCAL, s, v);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "127.0.0.1, true",
+        "localhost, true",
+        "svc.localhost, true",
+        "169.254.0.0, true",
+        "169.254.10.5, true",
+        "168.254.1.1, true",
+        "10.0.0.10, true",
+        "10.0.10.10, true",
+        "10.10.10.10, true",
+        "11.10.10.10, true",
+        "172.16.0.0, true",
+        "172.16.10.5, true",
+        "172.17.10.5, true",
+        "172.100.10.5, true",
+        "192.168.1.1, true",
+        "192.168.2.1, true",
+        "192.169.2.1, true",
+        "svc.local, true",
+        "example.svc.cluster.local, true",
+        "example.com, true"
+    })
+    public void testPublicRange(String s, String v) throws URISyntaxException {
+        test(URIRange.PUBLIC, s, v);
+    }
+
+    private void test(URIRange range, String s, String v) throws URISyntaxException {
+        boolean expected = Boolean.parseBoolean(v);
+        URI u = new URI(String.format("http://%s:1234", s));
```

Comment:
How about adding either a second assertion within this method, or a second similar method, that does `new URI(new JMXServiceURL(String.format("service:jmx:rmi:///jndi/rmi://%s:9091/jmxrmi", s).toString()))`, or something like that? Just to be certain that this utility accepts both JMX URLs and HTTP URLs, and also importantly that it not only parses them but that it treats them the same for a given URI range?

Reply:
seems like it does not work well with ` JMX URLs`

Reply:
What happens? Could you push the tests you have even if they're failing?